console/console-init/ui/src/modules/device-detail/components/CredentialsView/CredentialsView.tsx

```diff
@@ -0,0 +1,145 @@
+ */
+
+import React from "react";
+import {
+  Title,
+  Grid,
+  GridItem,
+  Card,
+  CardBody,
+  CardHeader,
+  Divider,
+  Text,
+  TextVariants
+} from "@patternfly/react-core";
+import { getLabelByKey } from "utils";
+import { SwitchWithToggle } from "components";
+import { SecretsView, ISecretsViewProps } from "./SecretsView";
+import { ExtensionsView } from "./ExtensionsView";
+import { hasOwnProperty } from "utils";
+import { StyleSheet } from "@patternfly/react-styles";
+
+const styles = StyleSheet.create({
+  row_margin: {
+    marginBottom: 5
+  },
+  status_section_margin: {
+    marginTop: 20,
+    marginBottom: 10
+  },
+  devider_margin: {
+    marginTop: 35,
+    marginBottom: 35
+  }
+});
+
+interface ICredentialView extends Pick<ISecretsViewProps, "secrets"> {
+  "auth-id"?: string;
+  type?: string;
+  ext?: any;
+  enabled?: boolean;
+}
+
+export interface ICredentialsViewProps {
+  id: string;
+  credentials: ICredentialView[];
+}
+
+export const Credential: React.FC<{ credential: ICredentialView }> = ({
+  credential
+}) => {
+  const { secrets = [], ext = [], type, enabled, "auth-id": authId } =
+    credential || {};
+  return (
+    <>
+      <Grid>
+        <GridItem span={3}>
+          <Title headingLevel="h1" size="md">
+            <b>{getLabelByKey("auth-id")}</b>
+          </Title>
+        </GridItem>
+        <GridItem span={9} className={styles.row_margin}>
+          {authId}
+        </GridItem>
+        <GridItem span={3}>
+          <Title headingLevel="h1" size="md">
+            <b>{getLabelByKey("type")}</b>
+          </Title>
+        </GridItem>
+        <GridItem span={9} className={styles.row_margin}>
+          {getLabelByKey(type || "")}
+        </GridItem>
+        <SecretsView
+          secrets={secrets}
+          id={"credetials-view-secrets-" + authId}
+          heading={"Secrets"}
+        />
+        <ExtensionsView
+          id={"credetials-view-extensions-" + authId}
+          ext={ext}
+          heading={"Ext"}
+        />
+        {hasOwnProperty(credential, "enabled") && (
+          <>
+            <GridItem span={12} className={styles.status_section_margin}>
+              Status
+            </GridItem>
+            <GridItem span={3}>
+              <Title headingLevel="h1" size="md">
+                <b>Enable</b>
+              </Title>
+            </GridItem>
+            <GridItem span={9}>
+              <SwitchWithToggle
+                id={authId}
+                label={"On"}
+                labelOff={"Off"}
+                isChecked={enabled}
+              />
+            </GridItem>
+          </>
+        )}
+      </Grid>
+    </>
+  );
+};
+
+export const CredentialsView: React.FC<ICredentialsViewProps> = ({
+  id,
+  credentials
+}) => {
+  const CredentialsNotFound = () => (
+    <Text component={TextVariants.p}>
+      There are no device for this device. This device is connected to the other
```

Comment:
device -> credentials

Reply:
I don't get you. are you talking about component name or something else ?

Reply:
I am talking about the line 116. According to the designs the sentence should be "There are no credentials for this ..."

Reply:
ok. I got it.